services/telemetry/src/telemetry.rs

```diff
@@ -67,11 +86,30 @@ pub async fn start_telemetry(state_ref: Arc<Mutex<State>>) -> io::Result<()> {
     
     let mut child_out = BufReader::new(decode_aprs.stdout.as_mut().unwrap());
     let mut readbuf = vec![0; 256];
+    let mut buffer = Vec::new();
 
     loop {
         // read() blocks until data is available. This makes this not polling.
-        let mut b = child_out.read(&mut readbuf);
-        extract_packets(Arc::clone(&state_ref), readbuf.clone()).await;
+        let mut bytes_read = child_out.read(&mut readbuf);
+        match bytes_read {
+            Ok(value) => {
+                let int_bytes_read: i32 = value as i32;
+                if (int_bytes_read > 0) {
+                    buffer.extend_from_slice(&readbuf[..value]);
+                    // Check for two successive newlines
+                    if let Some(pos) = buffer.windows(2).position(|window| window == b"\n\n") {
```

Comment:
when I run the code it couldn't find two consecutive new lines. We should test this with the brb on Thursday 

I did this before the if statement and it worked fine with the mock data
```rs
 extract_packets(Arc::clone(&state_ref), buffer.clone()).await;
 buffer.clear();
```

Reply:
@JackCotter Could we add a `Locked` attribute to the packet so we can more easily tell on the frontend if we've got a lock or not